### PR TITLE
cargo-bump: init at 1.1.1

### PR DESCRIPTION
**Pull request by cafkafk**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/rust/cargo-bump/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  };
+
+  cargoHash = "sha256-mp2y5q0GYfSlB5aPC6MY9Go8a2JAiPKtVYL9SewfloI=";
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+  ] ++ lib.optionals stdenv.isDarwin [
```
Comment (wegank) [inline pkgs/development/tools/rust/cargo-bump/default.nix:27]:
```suggestion
  buildInputs = lib.optionals stdenv.isDarwin [
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -17267,6 +17267,7 @@ with pkgs;
   cargo-binutils = callPackage ../development/tools/rust/cargo-binutils { };
   cargo-bloat = callPackage ../development/tools/rust/cargo-bloat { };
   cargo-bolero = callPackage ../development/tools/rust/cargo-bolero { };
+  cargo-bump = callPackage ../development/tools/rust/cargo-bump { };
```
Comment (wegank) [inline pkgs/top-level/all-packages.nix:17270]: Could the package be moved to `pkgs/by-name`?

Comment (cafkafk) [reply pkgs/top-level/all-packages.nix:17270]: My general understanding is that we shouldn't yet.



This is from yesterday.

File: pkgs/development/tools/rust/cargo-bump/default.nix (hunk 1 above)
Comment (cafkafk) [reply pkgs/development/tools/rust/cargo-bump/default.nix]: Yes, that's much cleaner, thanks!

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (wegank) [reply pkgs/top-level/all-packages.nix:17270]: For existing packages, no. But this one is an init, right?

Comment (cafkafk) [reply pkgs/top-level/all-packages.nix:17270]: Here's the [link](https://github.com/NixOS/nixpkgs/blob/f4822bb295ea194a1278af287771c615636cbcbf/pkgs/by-name/README.md#manual-migration-guidelines). It does seem to suggest that it should be initalized like that:

> This is the recommended way to add new top-level packages to Nixpkgs [when possible](https://github.com/NixOS/nixpkgs/blob/f4822bb295ea194a1278af287771c615636cbcbf/pkgs/by-name/README.md#limitations).

Do you want me to?

Comment (wegank) [reply pkgs/top-level/all-packages.nix:17270]: Yes, please.

Comment (cafkafk) [reply pkgs/top-level/all-packages.nix:17270]: I tried to use `by-name`, but seems like it will take a little extra work.

Comment (cafkafk) [reply pkgs/top-level/all-packages.nix:17270]: Okay, works!

**Comment by cafkafk** [comment]
Some builds fail because `https://github.com/rustadopt/cargo-bump/archive/1.1.1.tar.gz` should be `https://github.com/rustadopt/cargo-bump/archive/v1.1.1.tar.gz` it seems.

**Review by wegank — APPROVED** [review]

**Pull request merged**